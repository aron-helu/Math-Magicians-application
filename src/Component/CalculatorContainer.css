.container-calculator {
    height: 70vh;
    display: flex;
    justify-content: space-between;
}

.header-Cal {
    font-size: 25px;
    font-family: 'Open Sans', sans-serif;
    padding: 2px 20px;
}

.ar {
    width: 30%;
    padding: 60px 10px;
}

.card {
    width: 350px;
    height: 540px;
    padding: 10px;
    border-radius: 20px;
    background-color: rgba(189, 201, 216, 0.5);
    margin: auto 0;
}

.screen {
    height: 80px;
    width: 100%;
    margin-bottom: 15px;
    padding: 5px 10px;
    background-color: #8c989e2d;
    border-radius: 15px;
    display: flex;
    align-items: center;
    justify-content: flex-end;
    color: rgb(17, 16, 16);
    font-weight: bold;
    box-sizing: border-box;
}