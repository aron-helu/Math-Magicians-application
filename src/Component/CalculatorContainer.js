import React, { useState } from 'react';
import Calculator from './Calculator';
import calculate from '../logic/calculate';
import ButtonCard from '../UI/ButtonCard';
import './CalculatorContainer.css';
/* eslint-disable react/no-array-index-key */

const btnValues = [
  ['AC', '+/-', '%', 'รท'],
  ['7', '8', '9', 'x'],
  ['4', '5', '6', '-'],
  ['1', '2', '3', '+'],
  ['0', '.', '='],
];

const CalculatorContainer = () => {
  const [state, setState] = useState({
    total: null,
    next: null,
    operation: null,
  });
  const clickHandler = (event) => {
    const newState = calculate(state, event);
    setState((prevState) => ({
      ...prevState,
      ...newState,
    }));
  };
  return (
    <div className="container-calculator">
      <h2 className="header-Cal">Let&lsquo;s do some math!</h2>
      <div className="ar">
        <div className="card">
          <div className="screen">{state.next || state.total || 0}</div>
          <ButtonCard>
            {btnValues.flat().map((btn, i) => (
              <Calculator clickHandler={clickHandler} key={i} valueBtn={btn} />
            ))}
          </ButtonCard>
        </div>
      </div>
    </div>
  );
};

export default CalculatorContainer;
