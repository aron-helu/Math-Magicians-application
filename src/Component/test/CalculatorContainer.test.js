import renderer from 'react-test-renderer';
import CalculatorContainer from '../CalculatorContainer';

it('render correctly', () => {
    const tree = renderer.create(<CalculatorContainer />).toJSON();
    expect(tree).toMatchSnapshot();
});
