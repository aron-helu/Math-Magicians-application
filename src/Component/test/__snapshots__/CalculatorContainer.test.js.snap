// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render correctly 1`] = `
<div
  className="container-calculator"
>
  <h2
    className="header-Cal"
  >
    Let‘s do some math!
  </h2>
  <div
    className="ar"
  >
    <div
      className="card"
    >
      <div
        className="screen"
      >
        0
      </div>
      <div
        className="buttonBox"
      >
        <button
          onClick={[Function]}
          type="button"
        >
          AC
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          +/-
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          %
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          ÷
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          7
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          8
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          9
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          x
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          4
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          5
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          6
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          -
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          1
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          2
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          3
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          +
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          0
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          .
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          =
        </button>
      </div>
    </div>
  </div>
</div>
`;
